pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -810,10 +811,11 @@ func lookupOrAllocateIPv4(
 	log.Functionf("lookupOrAllocateIPv4(%s-%s): mac:%s\n",
 		status.DisplayName, status.Key(), mac.String())
 	// Lookup to see if it exists
-	if ip, ok := status.IPAssignments[mac.String()]; ok {
+	if ips, ok := status.IPAssignments[mac.String()]; ok && len(ips) != 0 {
 		log.Functionf("found Ip addr ( %s) for mac(%s)\n",
-			ip.String(), mac.String())
-		return ip.String(), nil
+			ips, mac.String())
+		ipAddr := ips[0]
```

Comment:
One remaining case of assuming [0] is IPv4.

It might be more clear to rename to IPv4Assignments and introduce a separate IPv6Assignments.